pkg/convert/testdata/programs/registry_module/pcl/main.pp

```diff
@@ -1,11 +1,11 @@
 component "cidrs" "./subnets_1.0.0" {
   baseCidrBlock = "10.0.0.0/8"
   networks = [{
-    name    = "foo"
-    newBits = 8
+    name     = "foo"
+    new_bits = 8
     }, {
-    name    = "bar"
-    newBits = 8
+    name     = "bar"
+    new_bits = 8
```

Comment:
Is networks here a list of maps?

Reply:
https://registry.terraform.io/modules/hashicorp/subnets/cidr/latest?tab=inputs

`networks list(object({ name = string new_bits = number }))`

object() looks like we should have marked this as _not_ a map. So I'll look into this.

Reply:
Yup so this is two problems
1. I screwed up and flipped the assumption of defaulting to rename if we didn't know the type.
2. We're not actually getting the type for this, even though we _probably_ could. I think we should merge this PR and then open a new one to look at seeing if we can write a module using maps and objects and check that translates correctly.

Reply:
> We're not actually getting the type for this, even though we probably could

We should get the type from the mapping

> we should merge this PR and then open a new one to look at seeing if we can write a module using maps and objects and check that translates correctly.

Sounds good!